pkg/client/unversioned/remotecommand/v1.go

```diff
@@ -28,19 +28,27 @@ import (
 )
 
 // streamProtocolV1 implements the first version of the streaming exec & attach
-// protocol. This version has some bugs, such as not being able to detecte when
+// protocol. This version has some bugs, such as not being able to detect when
 // non-interactive stdin data has ended. See http://issues.k8s.io/13394 and
 // http://issues.k8s.io/13395 for more details.
 type streamProtocolV1 struct {
-	stdin  io.Reader
-	stdout io.Writer
-	stderr io.Writer
-	tty    bool
+	StreamOptions
+
+	errorStream  httpstream.Stream
```

Comment:
Ok I remember why the v1 protocol has these as `httpstream.Stream` and not `io.Reader` etc - it's because the v1 protocol calls `Reset` on the streams. I'm updating v3 to use an `io.Writer` for `resizeStream`.